src/schema/v2/partnerArtworksMutation.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  GraphQLBoolean,
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLString,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import {
+  GravityMutationErrorType,
+  formatGravityError,
+} from "lib/gravityErrorHandler"
+import { GraphQLObjectType } from "graphql"
+import { InternalIDFields } from "./object_identification"
+import { GraphQLUnionType } from "graphql"
+interface Input {
+  id: string
+  artsyShippingDomestic: boolean | null
+  artsyShippingInternational: boolean | null
+  location: string | null
+}
+
+const PartnerArtworksType = new GraphQLObjectType<any, ResolverContext>({
+  name: "PartnerArtworks",
+  fields: () => ({
+    ...InternalIDFields,
+    success: { type: GraphQLInt },
+    errors: {
+      type: new GraphQLObjectType({
+        name: "PartnerArtworksError",
+        fields: {
+          count: { type: GraphQLInt },
+          ids: { type: GraphQLList(GraphQLString) },
+        },
+      }),
+    },
+  }),
+})
+
+const UpdatePartnerArtworksMutationSuccessType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "UpdatePartnerArtworksMutationSuccess",
+  isTypeOf: (data) => data.success || data.errors,
```

Comment:
Seems confusing, a `Success` type can be returned if the underlying data contains `errors`?

Reply:
I guess it can be confusing without some context. The REST endpoint response looks like so

```javascript
// All artworks updated
{
  "success": 4,
  "errors": {
    "count": 0,
    "ids": []
  }
}

// Some / no artworks updated
{
  "success": 0,
  "errors": {
    "count": 4,
    "ids": [
      "5ab3ea0b139b211428d47600",
      "5c7ecd5ecb5b32002879ff59",
      "5c7ecd5e53241d002c6b30c0",
      "5c7ecd5ef5464e269088ff9f"
    ]
  }
}
```

What Leandro and I found is that if we just put `isTypeOf: (data) => data.success` then it fails if `success` is zero. The `errors` will always be valid, even if empty, because it is an object which is inherently true, as opposed to `0` which can be interpreted as `false`. The endpoint successfully returns, but we can't know how many artworks (if any) will be updated ahead of time, due to other factors.

Reply:
Ah yea, if it's an integer and can be 0 as 'success', there's an `isExisty` helper (https://github.com/artsy/metaphysics/blob/d7bdf8300427c70aec30b8c55737770e5ebcb277/src/lib/helpers.ts#L33) you could use, probably `isExisty(data.success)`.

Reply:
Perfect, updated!